src/third-party/sonic/CMakeLists.txt

```
@@ -13,35 +13,60 @@
 
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+find_package(Sonic REQUIRED)
+set(SONIC_DIR "${CMAKE_SOURCE_DIR}/external/libs/sonic")
 
-add_library(sonic SHARED "${CMAKE_CURRENT_SOURCE_DIR}/sonic.c")
+if(SONIC_FOUND)
+	message(STATUS "Sonic found, no need to recompile, using the one from the packages of your distro. Set Sonic_FORCE_COMPILATION if you want to enforce recompilation")
```

i think this message still isn't correct.
its overly verbose and doesn't read as console output.
  
```suggestion
	message(STATUS "Sonic found, using from the distribution package. Set Sonic_FORCE_COMPILATION if you want to use it from a submodule.")
```

---

Thanks for 6he correction, this way is rewlly better. (but technically, it is from multiple packages, one for ueaders and another one for a shared library)

---

you forgot to apply my correction, didn't you?

---

Yeah, forgot to `git add`.